src/BaGet.UI/src/DisplayPackage/RepoDetails.tsx

```diff
@@ -0,0 +1,105 @@
+import { Icon } from 'office-ui-fabric-react/lib/Icon'
+import { Link } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+import './RepoDetails.css'
+
+interface IRepoDetailsProps {
+  url: string;
+}
+
+interface IRepoDetailsState {
+  openIssues: number;
+  openPulls: number;
+  lastCommit: Date;
+  detailsFound: boolean;
+}
+
+class RepoDetails extends React.Component<IRepoDetailsProps, IRepoDetailsState> {
+
+  private controller: AbortController;
+
+  constructor(props: IRepoDetailsProps) {
+    super(props);
+
+    this.controller = new AbortController();
+
+    this.state = {
+      openIssues: 0,
+      openPulls: 0,
+      lastCommit: new Date(),
+      detailsFound: false,
+    };
+  }
+
+  private getRepoOwnerAndName(repoUrl: string) {
+    const frags = repoUrl.split("/");
```

Comment:
We should check that the repository URL's hostname is GitHub. You should be able to use something like:

```ts
if (new URL(repoUrl).hostname === 'github.com')
```

See: https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname

Reply:
Also, you can split on the url's `pathname`: https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname

Reply:
Cheers for this, will use it 👍